thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftProtocolFactoryProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.thrift;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+
+import org.apache.thrift.protocol.TProtocolFactory;
+
+import com.linecorp.armeria.common.SerializationFormat;
+
+/**
+ * SPI Provider for links from {@link SerializationFormat} to {@link TProtocolFactory}.
+ */
+@FunctionalInterface
+public interface ThriftProtocolFactoryProvider {
+    /**
+     * Pair of {@link SerializationFormat} and {@link TProtocolFactory}.
+     */
+    final class ThriftSerializationFormat {
```

Comment:
This doesn't need to be visible to other classes than subclasses. How about making this class `protected` and `ThriftProtocolFactoryProvider` a public abstract class, like we did in `SerializationFormatProvider`?

Reply:
Subclasses and classes in same package (like `SerializationFormat` uses `SerializationFormatProvider.Entry`). ;)

I used interface to make the test class inherit multiple providers.

However, let me change to follow the existing pattern.

Reply:
Thanks!